qiskit/assertions/asserts.py

```diff
@@ -0,0 +1,120 @@
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, qubit, cbit, pcrit, negate, type_str):
+        """
+        Constructor for Asserts
+
+        Args:
+            qubit(QuantumRegister or list or tuple): quantum register
+            cbit(ClassicalRegister or list or tuple): classical register
+            pcrit(float): the critical p-value
+            negate(bool): True if assertion passed is negation of statistical test passed
+            type_str(string): the type of the assertion
+        """
+        super().__init__()
+        self._qubit = qubit
+        self._cbit = cbit
+        self._pcrit = pcrit
+        self._negate = negate
+        self._type = type_str
+        self._expval = None
+
+    @staticmethod
+    def breakpoint_name():
+        """
+        Returns the name of the breakpoint.  A breakpoint is a QuantumCircuit
+        whose last instruction is an assertion instruction.
+
+        Returns:
+            string: the breakpoint name
+        """
+        return "breakpoint_" + datetime.now().isoformat()
+
+    @staticmethod
+    def syntax4measure(bit):
+        """
+        Returns the input qubit or cbit, suitable for the syntax of the measure
+        instruction, as a list.
+
+        Args:
+            bit(QuantumRegister or ClassicalRegister or list or tuple): register
+
+        Returns:
+            list or Register: bit as a list
+        """
+        if isinstance(bit, (list, Register)):
+            return bit
+        elif isinstance(bit, (range, tuple)):
+            return list(bit)
+        else:
+            return [bit]
+
+    @staticmethod
+    def clbits2idxs(cbits, exp):
+        """
+        Returns the indices of cbits with respect to exp, the QuantumCircuit.
+
+        Args:
+            cbits(ClassicalRegister or list or tuple): cbit of a measurement instruction
+            exp(QuantumCircuit): an experiment
+
+        Returns:
+            list or range: the indices
+
+        Raises:
+            QiskitError: Invalid syntax for cbit
+        """
+        if isinstance(cbits[0], int):
+            return cbits
+        elif isinstance(cbits[0], Clbit):
+            idxs = [exp.clbits.index(cbit) for cbit in cbits]
+            return idxs
+        elif isinstance(cbits, ClassicalRegister):
+            idxfirst = exp.clbits.index(cbits[0])
+            idxlast = exp.clbits.index(cbits[-1])
+            return range(idxfirst, idxlast+1)
+        else:
+            raise QiskitError("Invalid syntax for cbit")
+
+    @abc.abstractmethod
+    def stat_test(self, counts):
+        """
+        Abstract method which performs a statistical test on the experimental outcomes
+
+        Args:
+            counts(dictionary): result.get_counts(experiment)
+
+        Returns:
+            tuple: tuple containing:
+
+                chisq(float): the chi-squared value
+                pval(float): the p-value
+                passed(bool): if the test passed
+        """
+        return
```

Comment:
maybe `raise NotImplementedError`?

Reply:
[resolved by this commit](https://github.com/Qiskit/qiskit-terra/pull/2867/commits/e64d7ea155c224da82759685a0c44e59aa11758a)